drivers/digital-io/max22200/max22200.c

```diff
@@ -0,0 +1,958 @@
+
+	ret = max22200_reg_read(desc, MAX22200_CFG_CH(ch), &reg_val, true);
+	if (ret)
+		return ret;
+
+	*enabled =  no_os_field_get(MAX22200_CH_ENABLE_MASK(ch_enable),
+				    reg_val);
+
+	return 0;
+}
+
+/**
+ * @brief Read configuration for the detection of the plunger movement of the
+ * 	  device
+ * @param desc - MAX22200 device descriptor.
+ * @param dpm_istart_byte - Detection of plunger movement starting current
+ * 			    value.
+ * @param dpm_tdeb_ipth_byte - Detection of plunger movement debounce time
+ * 			       value.
+ * @return 0 in case of succes, negative error code otherwise.
+*/
+int max22200_get_cfg_dpm(struct max22200_desc *desc, uint8_t *dpm_istart_byte,
+			 uint8_t *dpm_tdeb_ipth_byte)
+{
+	int ret;
+	uint32_t reg_val;
+
+	ret = max22200_reg_read(desc, MAX22200_CFG_DPM_REG, &reg_val, false);
+	if (ret)
+		return ret;
+
+	*dpm_istart_byte = no_os_field_get(MAX22200_DPM_ISTART_MASK, reg_val);
+	*dpm_tdeb_ipth_byte = no_os_field_get(MAX22200_DPM_TDEB_MASK, reg_val) |
+			      no_os_field_get(MAX22200_DPM_IPTH_MASK, reg_val);
+
+	return 0;
+}
+
+/**
+ * @brief MAX22200 descriptor initialization function.
+ * @param desc - MAX22200 device descriptor.
+ * @param init_param - Initialization parameter containing data about the device
+ * 		       descriptor to be initialized.
+ * @return 0 in case of succes, negative error code otherwise.
+*/
+int max22200_init(struct max22200_desc **desc,
+		  struct max22200_init_param *init_param)
+{
+	struct max22200_desc *descriptor;
+	uint32_t reg_val;
+	uint32_t status_reg = 0;
+	int ret, i;
+
+	descriptor = no_os_calloc(1, sizeof(*descriptor));
+	if (!descriptor)
+		return -ENOMEM;
+
+	ret = no_os_spi_init(&descriptor->comm_desc, init_param->comm_param);
+	if (ret)
+		goto err;
+
+	/* Time between enable pin set and device power-up. */
+	no_os_udelay(500);
+
+	ret = no_os_gpio_get(&descriptor->cmd_desc,
+			     init_param->cmd_param);
+	if (ret)
+		goto spi_err;
+
+	ret = no_os_gpio_direction_output(descriptor->cmd_desc, NO_OS_GPIO_LOW);
+	if (ret)
+		goto cmd_err;
+
+	/* Enable pin is optional to be connected to master through GPIO and be
+	   controlled in this way, but mandatory to be logic high (3.3V) in
+	   order for the device to be enabled. */
+	ret = no_os_gpio_get_optional(&descriptor->enable_desc,
+				      init_param->enable_param);
+	if (ret)
+		goto cmd_err;
+
+	if (descriptor->enable_desc) {
+		ret = no_os_gpio_direction_output(descriptor->enable_desc,
+						  NO_OS_GPIO_HIGH);
+		if (ret)
+			goto enable_err;
+	}
+
+	ret = no_os_gpio_get_optional(&descriptor->trig_desc,
+				      init_param->trig_param);
+	if (ret)
+		goto enable_err;
+
+	/* External trigger if used, is set logic LOW at initialization.
+	   Can be set logic HIGH in case of choosing to use the external
+	   trigger for Full-Bridge Mode. */
+	if (descriptor->trig_desc) {
+		ret = no_os_gpio_direction_output(descriptor->trig_desc,
+						  NO_OS_GPIO_LOW);
+		if (ret)
+			goto trig_err;
```

Comment:
enable_err, trig_desc is not allocated in this case

Reply:
the if condtion above checks if trig_desc is initialized there by allocated, same for the fault_err handling.

Reply:
oops, my bad. I misread the code.